webconsole/v1/types.go

```diff
@@ -13,55 +13,51 @@ type WebConsoleConfiguration struct {
 	// ServingInfo is the HTTP serving information for these assets
 	ServingInfo HTTPServingInfo `json:"servingInfo" protobuf:"bytes,1,opt,name=servingInfo"`
 
-	// PublicURL is where you can find the asset server (TODO do we really need this?)
+	// PublicURL is where you can find the web console server (TODO do we really need this?)
 	PublicURL string `json:"publicURL" protobuf:"bytes,2,opt,name=publicURL"`
 
-	// LogoutURL is an optional, absolute URL to redirect web browsers to after logging out of the web console.
-	// If not specified, the built-in logout page is shown.
-	LogoutURL string `json:"logoutURL" protobuf:"bytes,3,opt,name=logoutURL"`
+	// InactivityTimeoutMinutes is the number of minutes of inactivity before you are automatically logged out of
+	// the web console (optional). If set to 0, inactivity timeout is disabled.
+	InactivityTimeoutMinutes int64 `json:"inactivityTimeoutMinutes" protobuf:"varint,3,opt,name=inactivityTimeoutMinutes"`
 
-	// MasterPublicURL is how the web console can access the OpenShift v1 server
-	MasterPublicURL string `json:"masterPublicURL" protobuf:"bytes,4,opt,name=masterPublicURL"`
+	// ClusterInfo holds information the web console needs to talk to the cluster such as master public URL
+	// and metrics public URL
+	ClusterInfo ClusterInfo `json:"clusterInfo" protobuf:"bytes,4,rep,name=clusterInfo"`
 
-	// LoggingPublicURL is the public endpoint for logging (optional)
-	LoggingPublicURL string `json:"loggingPublicURL" protobuf:"bytes,5,opt,name=loggingPublicURL"`
+	// Extensions define custom scripts, stylesheets, and properties used for web console customization
+	Extensions ExtensionsConfiguration `json:"extensions" protobuf:"bytes,5,rep,name=extensions"`
+}
 
-	// MetricsPublicURL is the public endpoint for metrics (optional)
-	MetricsPublicURL string `json:"metricsPublicURL" protobuf:"bytes,6,opt,name=metricsPublicURL"`
+// ClusterInfo holds information the web console needs to talk to the cluster such as master public URL and
+// metrics public URL
+type ClusterInfo struct {
+	// MasterPublicURL is how the web console can access the OpenShift v1 server
+	MasterPublicURL string `j4on:"masterPublicURL" protobuf:"bytes,1,opt,name=masterPublicURL"`
```

Comment:
j4on?

Reply:
> j4on?

@sttts is there a linter of a test we have that would catch stuff like this?

Reply:
j4on is valid tag name, isn't it?

Reply:
> j4on is valid tag name, isn't it?

Valid tag name, but there should be a json tag for every external field that doesn't explicitly opt out.